### PR TITLE
[Form] Documented the intl option in TimezoneType

### LINKED ISSUE #11503: [Form] Add intl/choice_translation_locale options to TimezoneType
see https://github.com/symfony/symfony/pull/31294 (ref https://github.com/symfony/symfony-docs/issues/9657)

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #11503.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@ro0NL if you have some time, please review this. Thanks!

File: reference/forms/types/timezone.rst [hunk 1]
```diff
@@ -105,6 +129,8 @@ Inherited Options
 
 These options inherit from the :doc:`ChoiceType </reference/forms/types/choice>`:
 
+.. include:: /reference/forms/types/options/choice_translation_domain.rst.inc
```
Comment (xabbuh) [inline reference/forms/types/timezone.rst]: I think we should add a note that this option can only be used when `intl` is set to `true`.

Comment (javiereguiluz) [reply reference/forms/types/timezone.rst]: I thought about that ... however, I saw that the code shows a perfect error message when you do that (see https://github.com/symfony/symfony/pull/31294/files#diff-c6198934d4f35bc102ff362d40b71285R72) so I guess it's fine to not add these implementation details in the docs.

Comment (ro0NL) [reply reference/forms/types/timezone.rst]: It's also more or less explained above. Works for me :)

**Review by ro0NL — APPROVED** [review]

File: reference/forms/types/timezone.rst (hunk 1 above)
Comment (xabbuh) [reply reference/forms/types/timezone.rst]: My thought is that people will just notice this option and try to use it. But it's not a big deal. We can add something later anyway if we receive complaints about it not being straight-forward (the error provided by the form type is quite clear indeed).

**Review by xabbuh — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Pull request merged**